core/services/eth/log_broadcaster.go

```diff
@@ -379,3 +381,12 @@ func (l *decodingLogListener) HandleLog(log interface{}, err error) {
 
 	l.LogListener.HandleLog(decodedLog, nil)
 }
+
+func appendLogChannel(oldLogChannel, newLogChannel chan eth.Log) chan eth.Log {
+	go func() {
```

Comment:
Hey, I'm not sure this is going to do what you want.  There's a possibility here that logs from `newLogChannel` will become interspersed with the last few logs from the old subscription.  That's why in my example I explicitly used two for loops and created a brand new channel to dump everything into.